lib/impress.files.js

```diff
@@ -31,77 +31,80 @@ Client.prototype.download = function(
   // Download file
   filePath, // file to download
   attachmentName, // name to save downloaded file, optional
-  callback // function
+  done // optional function
 ) {
   const client = this;
   const application = client.application;
 
   if (typeof(attachmentName) === 'function') {
-    callback = attachmentName;
+    done = attachmentName;
     attachmentName = api.path.basename(filePath);
   }
+
+  const fail = () => {
+    application.log.error(impress.CANT_READ_FILE + filePath);
+    client.error(404);
+    if (done) done();
+  };
+
+  const success = () => {
+    if (done) done();
+  };
+
   api.fs.stat(filePath, (err, stats) => {
-    if (err) {
-      client.error(404);
-      if (callback) callback();
-      return;
-    }
+    if (err) return fail();
     client.attachment(attachmentName, stats.size, stats.mtime.toGMTString());
     const stream = api.fs.createReadStream(filePath);
-    stream.on('error', () => {
-      application.log.error(impress.CANT_READ_FILE + filePath);
-      client.error(404);
-      if (callback) callback();
-    });
-    client.res.on('finish', () => {
-      if (callback) callback();
-    });
+    stream.on('error', fail);
+    client.res.on('finish', success);
     stream.pipe(client.res);
   });
 };
 
 Client.prototype.upload = function(
   // Upload file
-  onFile, // optional callback function on processing each file
+  each, // optional callback function on processing each file
   // { compressionFlag, originalName, storageName
   //   storagePath, originalHash, originalSize, storageSize }
-  onDone // optional callback function(doneCount)
+  done // optional callback function(doneCount)
 ) {
   const client = this;
 
   let fileCount = 0;
   let doneCount = 0;
 
-  const done = (data) => {
+  const cb = (data) => {
     doneCount++;
-    if (onFile) onFile(data);
-    if (fileCount === doneCount && onDone) onDone(doneCount);
+    if (each) each(data);
+    if (fileCount === doneCount && done) done(doneCount);
   };
 
-  if (client.files) {
-    let fieldName, key, field, file;
-    for (fieldName in client.files) {
-      field = client.files[fieldName];
-      for (key in field) {
-        file = field[key];
-        fileCount++;
-        client.uploadFile(file, done);
-      }
+  if (!client.files) {
+    if (done) return done(0);
+  }
+
+  let fieldName, key, field, file;
+  for (fieldName in client.files) {
+    field = client.files[fieldName];
+    for (key in field) {
+      file = field[key];
+      fileCount++;
+      client.uploadFile(file, cb);
     }
   }
 };
 
 function saveUploadedFile(
   // Save uploaded file
   data, // { compressionFlag, storagePath, storageSize }
-  onFile // function
+  each // function
```

Comment:
I guess `each` isn't a very appropriate name since there's only one file, not a collection of them.